essentials.sh

```diff
@@ -174,6 +174,26 @@ case $input in
  ;;
 esac
 
+
+read -r -p $'\nInstall Docker? [Y/n] ' input
+input=${input:-Y}
+case $input in
+    [yY][eE][sS]|[yY])
+    printf "\n\033[0;32mInstalling Docker\033[0m\n"    
+    sudo apt install docker.io
```

Comment:
Older versions of docker binary were called `docker.io`. The newer versions are called `docker-ce` and provides certified release.
I have not worked with docker but just for being future safe, I'd suggest to follow the official installation guide [here](https://docs.docker.com/engine/install/debian/).

Also there have been multiple issue of docker.io repositories being out of date.

Reply:
okey, just to confirm, I need to put the GPG key configs as well right?

Reply:
The main idea is to install `docker-ce` over `docker.io`, since docker.io package in the apt repository is maintained by Ubuntu/Debian and are most of the times outdated, while `docker-ce` provides the official updated package.
The preferred way to go would be to install docker-ce using Step-1 as mentioned in this [blog](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-18-04).

To make it work for Ubuntu 16.04 and 20.04 as well, we just need to change the version in command
`sudo add-apt-repository "deb [arch=amd64] https://download.docker.com/linux/ubuntu bionic stable"` from `bionic`(Ubuntu 18.04) to `xenial` (Ubuntu 16.04) or `focal` (Ubuntu 20.04).

You can get the ubuntu version from the command 
```bash
lsb_release -a 2>/dev/null| grep Codename | cut -f2 | awk '{print tolower($0)}'
```
You can store the output in a variable and then use it in the `add-apt-repository` step to get the correct repository. Hope this helps :smile: